### PR TITLE
Validate HF config keys before conversion

### PR DESCRIPTION
**Pull request by shadeMe**
## Description

Add checks to ensure that expected keys are always found in the HF `config.json` files when loading models from the HF Hub.

### Types of change

Enhancement

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: curated_transformers/models/albert/_hf.py [hunk 1]
```diff
@@ -19,27 +19,34 @@
...
-        num_hidden_groups=hf_config["num_hidden_groups"],
-        num_hidden_layers=hf_config["num_hidden_layers"],
-        padding_id=padding_id,
-        type_vocab_size=hf_config["type_vocab_size"],
-        vocab_size=hf_config["vocab_size"],
+    missing_keys = tuple(
+        sorted(set(HF_CONFIG_KEY_MAPPING.keys()).difference(set(hf_config.keys())))
     )
+    if len(missing_keys) != 0:
+        raise ValueError(f"Missing keys in HF AlBERT model config: {missing_keys}")
```
Comment (danieldk) [inline curated_transformers/models/albert/_hf.py:46]:
```suggestion
        raise ValueError(f"Missing keys in HF ALBERT model config: {missing_keys}")
```

File: curated_transformers/models/refined_web_model/_hf.py [hunk 2]
```diff
@@ -11,29 +11,39 @@
...
+    "vocab_size": "vocab_size",
+}
+
+
 def convert_hf_config(hf_config: Any) -> RefinedWebModelConfig:
+    missing_keys = tuple(
+        sorted(set(HF_CONFIG_KEY_MAPPING.keys()).difference(set(hf_config.keys())))
+    )
+    if len(missing_keys) != 0:
+        raise ValueError(f"Missing keys in HF RefinedWeb model config: {missing_keys}")
```
Comment (danieldk) [inline curated_transformers/models/refined_web_model/_hf.py:30]:
```suggestion
        raise ValueError(f"Missing keys in Hugging Face Refined Web Model config: {missing_keys}")
```

**Review by danieldk — COMMENTED** [review]
This is really a nice improvement!

Some requested doc fixes. Feel free to merge afterwards.

File: curated_transformers/models/albert/_hf.py (hunk 1 above)
Comment (shadeMe) [reply curated_transformers/models/albert/_hf.py:46]: Good point. I'll add that to my upcoming docstring PR.

File: curated_transformers/models/roberta/_hf.py [hunk 3]
```diff
@@ -17,24 +17,36 @@
...
         # Positions embeddings for 0..padding_id are reserved.
-        model_max_length=hf_config["max_position_embeddings"] - (padding_id + 1),
-        max_position_embeddings=hf_config["max_position_embeddings"],
-        num_attention_heads=hf_config["num_attention_heads"],
-        num_hidden_layers=hf_config["num_hidden_layers"],
-        padding_id=padding_id,
-        type_vocab_size=hf_config["type_vocab_size"],
-        vocab_size=hf_config["vocab_size"],
+        model_max_length=hf_config["max_position_embeddings"]
+        - (kwargs["padding_id"] + 1),
```
[the remark this answers is not in the document]
Comment (shadeMe) [reply curated_transformers/models/roberta/_hf.py:48]: Yup, that's `black`'s doing.

**Pull request merged**